lib/avrora/encoder.ex

```diff
@@ -90,7 +90,7 @@ defmodule Avrora.Encoder do
       ...> Avrora.Encoder.decode_plain(payload,"io.confluent.NumericTransfer")
       {:ok, %{"link_is_enabled" => false, "updated_at" => 1586632500, "updated_by_id" => 100}
   """
-  @spec decode_plain(binary(), schema_name: String.t()) :: {:ok, map()} | {:error, term()}
+  @spec decode_plain(binary(), schema_name: String.t()) :: {:ok, :avro.out()} | {:error, term()}
```

Comment:
I have double-checked the named types and it turns out that I misunderstand few things. So named types are:

> Record, enums and fixed are named types. Each has a fullname that is composed of two parts; a name and a namespace. Equality of names is defined on the fullname.

Only they can be used on a root level because the `erlavro` requires a named type to be the root. All 3 types will return either `String.t()` or `map()` type. I think to enable broader output typespec, the `is_map` guard should be fixed first for `encode_plain/2`.

I'm afraid it's a bit more work.